src/unitxt/metrics.py

```diff
@@ -82,43 +83,100 @@ def _can_compute_confidence_intervals(self, num_predictions):
             and num_predictions > 1
         )
 
-    def score_based_confidence_interval(self, instances):
-        """Compute confidence intervals based on existing scores, already computed on the input instances.
+    @staticmethod
+    def aggregate_instance_scores(instances, field_name):
+        """Calculate mean of a set of instance scores (given by field_name), ignoring NaNs.
 
-        score_names: List[str]
-            Compute a confidence interval for each score_name from this list.
-        instances:
-            The instances for which the confidence intervals are computed.
+        Args:
+            instances: The instances for which the confidence intervals are computed; should already have the relevant instance scores calculated.
+            field_name: score field names to compute mean for.
         """
-        from statistics import mean
+        scores = [instance["score"]["instance"][field_name] for instance in instances]
+        import warnings
+
+        with warnings.catch_warnings():
+            # in case instances is empty, return NaN but avoid printing a RuntimeWarning
+            warnings.simplefilter("ignore", category=RuntimeWarning)
+            return np.nanmean(scores)
```

Comment:
okay, I will return it to what it was before.  Previously, there was no Nanmean here, it was just mean.  If there are any NaN instance scores this will return NaN.  But I don't know any case where there are NaN instance scores.